awx/ui/test/unit/components/stream.unit.js

```diff
@@ -0,0 +1,101 @@
+import StreamService from '~features/output/stream.service';
+
+describe('Output | StreamService', () => {
+  angular.module('test', []).service('StreamService', StreamService);
+  let stream;
+
+  beforeEach(() => {
+    angular.mock.module('test');
+  });
+
+  beforeEach(angular.mock.inject(($injector) => {
+    stream = $injector.get('StreamService');
+
+    const onFrames = angular.noop;
+    const onFrameRate = angular.noop;
+
+    stream.init({ onFrames, onFrameRate });
+  }));
+
+  describe('calcFactors', () => {
+    it('returns the expected values', () => {
+      const params = [
+        [-1, [1]],
+        [0, [1]],
+        [1, [1]],
+        [1.0, [1]],
+        [1.1, [1]],
+        [2, [1, 2]],
+        ['1', [1]],
+        [{}, [1]],
+        [null, [1]],
+        [undefined, [1]],
+        [250, [1, 2, 5, 10, 25, 50, 125, 250]]
+      ];
+
+      params.forEach(([size, expected]) => expect(stream.calcFactors(size)).toEqual(expected));
+    });
+  });
+
+  describe('setMissingCounterThreshold', () => {
+    it('returns the correct counter threshold', () => {
+      const gt = 2;
+      stream.setMissingCounterThreshold(gt);
+      expect(stream.counters.min).toEqual(gt);
+
+      const lt = -1;
+      stream.setMissingCounterThreshold(lt);
+      expect(stream.counters.min).toEqual(gt);
+    });
+  });
+
+  describe('isReadyToRender', () => {
+    it('returns false', () => {
+      const res = stream.isReadyToRender();
+      expect(res).toBe(false);
+    });
+
```

Comment:
we talked about this one offline - we'd like to expand these test cases into something that captures the following behavior:

- “it's never ready to render unless the result of getReadyCount is greater than 0”
- "it's always ready to render if any events are ready and the stream is ending"